vuvuzela_test.go

```diff
@@ -0,0 +1,619 @@
+	}
+	if err := addFriendServer.Run(); err != nil {
+		log.Panicf("starting addfriend loop: %s", err)
+	}
+
+	u.DialingConfig = &config.SignedConfig{
+		Version: config.SignedConfigVersion,
+		Created: time.Now(),
+		Expires: time.Now().Add(24 * time.Hour),
+
+		Service: "Dialing",
+		Inner: &config.DialingConfig{
+			Version: config.DialingConfigVersion,
+			Coordinator: config.CoordinatorConfig{
+				Key:     coordinatorKey,
+				Address: coordinatorAddr,
+			},
+			MixServers: mixchain.Servers,
+			CDNServer: config.CDNServerConfig{
+				Key:     CDN.PublicKey,
+				Address: CDN.Addr,
+			},
+		},
+	}
+	err = u.ConfigServer.SetCurrentConfig(u.DialingConfig)
+	if err != nil {
+		log.Panicf("error setting current dialing config: %s", err)
+	}
+
+	dialingServer := &alpenhornCoordinator.Server{
+		Service:    "Dialing",
+		PrivateKey: coordinatorPrivate,
+		Log: log.WithFields(log.Fields{
+			"tag":     "coordinator",
+			"service": "Dialing",
+		}),
+
+		ConfigClient: u.ConfigClient,
+
+		MixWait:      1 * time.Second,
+		RoundWait:    2 * time.Second,
+		NumMailboxes: 1,
+
+		PersistPath: filepath.Join(u.AlpenhornDir, "dialing-coordinator-state"),
+	}
+	if err := dialingServer.Persist(); err != nil {
+		log.Panicf("error persisting dialing server: %s", err)
+	}
+	if err := dialingServer.LoadPersistedState(); err != nil {
+		log.Panicf("error loading persisted state: %s", err)
+	}
+	if err := dialingServer.Run(); err != nil {
+		log.Panicf("starting dialing loop: %s", err)
+	}
+
+	mux := http.NewServeMux()
+	mux.Handle("/addfriend/", http.StripPrefix("/addfriend", addFriendServer))
+	mux.Handle("/dialing/", http.StripPrefix("/dialing", dialingServer))
+	coordinatorHTTPServer := &http.Server{
+		Handler: mux,
+	}
+	go func() {
+		err := coordinatorHTTPServer.Serve(coordinatorListener)
+		if err != http.ErrServerClosed {
+			log.Fatalf("http.Serve: %s", err)
+		}
+	}()
+}
+
+// createUser creates a test client. It is based on main.generateVuvuzelaClient.
+func (u *universe) createUser(name string) *Client {
+
+	client := &Client{
+		PersistPath:        filepath.Join(u.VuvuzelaDir, fmt.Sprintf("%s-vuvuzela-client-state", name)),
+		ConfigClient:       u.ConfigClient,
+		Handler:            u.Handler,
+		CoordinatorLatency: 150 * time.Millisecond,
+	}
+	err := client.Bootstrap(u.ConvoConfig)
+	if err != nil {
+		log.Panicf("bootstrapping vuvuzela client: %s", err)
+	}
+	err = client.Persist()
+	if err != nil {
+		log.Panicf("persisting vuvuzela client: %s", err)
+	}
+	return client
+}
+
+type alpenhornClientHandler struct {
+	errPrefix string
+
+	confirmedFriend       chan *alpenhorn.Friend
+	sentFriendRequest     chan *alpenhorn.OutgoingFriendRequest
+	receivedFriendRequest chan *alpenhorn.IncomingFriendRequest
+	sentCall              chan *alpenhorn.OutgoingCall
+	receivedCall          chan *alpenhorn.IncomingCall
+	newConfig             chan []*config.SignedConfig
+}
+
+func newAlpenhornHandler(errPrefix string) *alpenhornClientHandler {
```

Comment:
Since this alpenhorn handler stuff is the same as in alpenhorn_test, should it be moved to internal/mock? It clutters up this file.